src/test/java/org/apache/commons/lang3/time/DurationFormatUtilsTest.java

```diff
@@ -632,4 +632,31 @@ public void testLowDurations() {
         }
     }
 
+    @Test
+    public void testUnmatchedOptionalToken() {
+        assertThrows(IllegalArgumentException.class, () -> DurationFormatUtils.formatDuration(1, "[s"));
```

Comment:
Nesting has little value that I can see, so simplest solution will be to catch any nesting and throw an IllegalArgumentException. That will prevent unbalancing issues since the current optional state is tracked with a boolean, similar to literals, and we can easily detect attempts to nest.

Multiple optional blocks are supported by default (though realistic use cases may be few). If you wanted to always show minutes, but omit zero-value hours and seconds, then the format `[H'h ']m'm'[s's ']` will produce values like:
```
254h16m
15m15s 
0m9s
```
Will add a commit to address the nesting issue.